RickLeinecker/Fortuna
# Refactored API calls to reduce code

**Description**
Refactored API calls (mainly in armory and marketplace) to remove code. This took a lot of time since a lot of the old code was very messy. The only ones I didn't refactor were the login and signup API calls, they are only used once (I could have probably done the same with marketplace).

Added in backend handling of getFavorite, when favorite tank is null.

**Resolves These Issues**
Resolves #309 
Resolves #412 

**Type of change**
Check options that are relevant.


**Steps to Verify Functionality**
1. Make a component sale and tank sale.
2. Ensure there were no errors when making the sales.
3. Purchase on another account.
4. Make sure favorite tank doesn't show up in the possible tanks to sell (was broken).
5. Go to Armory.
6. Do all tank options (rename, create, delete, setup, and remove).
7. Open component menu and ensure all components show up and you can swap.
8. Switch between selected tanks.

**Final Checklist:**
Go down the list and check them off.
